tests/webapp/api/test_performance_data_api.py

```diff
@@ -582,6 +588,252 @@ def test_perf_summary(client, test_perf_signature, test_perf_data):
+        suite='a11yr',
+        test='dhtml.html',
+        has_subtests=test_perf_signature.has_subtests,
+        extra_options="e10s fission stylo webrender",
+        last_updated=datetime.datetime.now(),
+        measurement_unit='ms',
+    )
+
+    for index, job in enumerate(perf_jobs[3:5]):
+        job.push = push2
+        job.save()
+
+        perf_datum = PerformanceDatum.objects.create(
+            value=sig2_values[index],
+            push_timestamp=job.push.time,
+            job=job,
+            push=job.push,
+            repository=job.repository,
+            signature=sig2,
+        )
+        perf_datum.push.time = job.push.time
+        perf_datum.push.save()
+
+    sig3 = PerformanceSignature.objects.create(
+        repository=test_perf_signature.repository,
+        signature_hash=(20 * 't3'),
+        framework=test_perf_signature.framework,
+        platform=macosx1015_64_shippable_qr,
+        option_collection=test_perf_signature.option_collection,
+        suite='a11yr',
+        test='dhtml.html',
+        has_subtests=test_perf_signature.has_subtests,
+        extra_options="e10s fission stylo webrender",
+        last_updated=datetime.datetime.now(),
+        measurement_unit='ms',
+    )
+
+    for index, job in enumerate(perf_jobs[5:7]):
+        job.push = push1
+        job.save()
+
+        perf_datum = PerformanceDatum.objects.create(
+            value=sig3_values[index],
+            push_timestamp=job.push.time,
+            job=job,
+            push=job.push,
+            repository=job.repository,
+            signature=sig3,
+        )
+        perf_datum.push.time = job.push.time
+        perf_datum.push.save()
+
+    sig4 = PerformanceSignature.objects.create(
+        repository=test_perf_signature.repository,
+        signature_hash=(20 * 't4'),
+        framework=test_perf_signature.framework,
+        platform=macosx1015_64_shippable_qr,
+        option_collection=test_perf_signature.option_collection,
+        suite='a11yr',
+        test='dhtml.html',
+        has_subtests=test_perf_signature.has_subtests,
+        extra_options="e10s fission stylo webrender",
+        last_updated=datetime.datetime.now(),
+        measurement_unit='ms',
+    )
+
+    for index, job in enumerate(perf_jobs[7:9]):
+        job.push = push2
+        job.save()
+
+        perf_datum = PerformanceDatum.objects.create(
+            value=sig4_values[index],
+            push_timestamp=job.push.time,
+            job=job,
+            push=job.push,
+            repository=job.repository,
+            signature=sig4,
+        )
+        perf_datum.push.time = job.push.time
+        perf_datum.push.save()
+
+    option_collection = OptionCollection.objects.select_related('option').values(
+        'id', 'option__name'
+    )
+    option_collection_map = {item['id']: item['option__name'] for item in list(option_collection)}
+
+    expected = [
+        {
+            'framework_id': sig1.framework.id,
```

Comment:
[question]
I saw we're missing signature_id and signature_hash

Reply:
I think that we may need some type of identifier like this for each result

Reply:
The `signature_id` and `signature_hash ` will not be displayed in the Results View and this is the reason I've excluded them. Other fields like the header name and platform can be used as identifiers. If we decide these fields are necessary we can add them later.